src/server/bitops_family.cc

```diff
@@ -685,6 +748,90 @@ OpResult<std::size_t> CountBitsForValue(const OpArgs& op_args, std::string_view
   }
 }
 
+// Returns the bit position (where MSB is 0, LSB is 7) of the leftmost bit that
+// equals `value` in `byte`. Returns 8 if not found.
+std::size_t GetFirstBitWithValueInByte(uint8_t byte, bool value) {
+  if (value) {
+    return absl::countl_zero(byte);
+  } else {
+    return absl::countl_one(byte);
+  }
+}
+
+int64_t FindFirstBitWithValueAsBit(std::string_view result_value, bool bit_value, int64_t start,
+                                   int64_t end) {
+  for (int64_t i = start; i <= end; ++i) {
+    if (static_cast<size_t>(GetByteIndex(i)) >= result_value.size()) {
+      break;
+    }
+    const uint8_t current_byte = GetByteValue(result_value, i);
+    bool current_bit = CheckBitStatus(current_byte, GetNormalizedBitIndex(i));
+    if (current_bit != bit_value) {
+      continue;
+    }
+
+    return i;
+  }
+
+  return -1;
+}
+
+int64_t FindFirstBitWithValueAsByte(std::string_view result_value, bool bit_value, int64_t start,
+                                    int64_t end) {
+  for (int64_t i = start; i <= end; ++i) {
+    if (static_cast<size_t>(i) >= result_value.size()) {
+      break;
+    }
+    const uint8_t current_byte = result_value[i];
+    const uint8_t kNotFoundByte = bit_value ? 0 : std::numeric_limits<uint8_t>::max();
+    if (current_byte == kNotFoundByte) {
+      continue;
+    }
+
+    return i * OFFSET_FACTOR + GetFirstBitWithValueInByte(current_byte, bit_value);
+  }
+
+  return -1;
+}
+
+OpResult<int64_t> FindFirstBitWithValue(const OpArgs& op_args, std::string_view key, bool bit_value,
+                                        int64_t start, int64_t end, bool as_bit) {
+  OpResult<std::string> result = ReadValue(op_args.db_cntx, key, op_args.shard);
+
+  std::string_view result_value;
```

Comment:
Same here, could be input_value or similar.

Reply:
I went with `value_str` to reflect that it's the value's string value. lmk if you have a better idea.